pkgs/by-name/tu/turborepo-remote-cache/package.nix

```
@@ -0,0 +1,73 @@
+{
+  lib,
+  stdenvNoCC,
+  nodejs-slim_22,
+  pnpm_10,
+  jq,
+  fetchFromGitHub,
+  makeWrapper,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "turborepo-remote-cache";
+  version = "2.6.0";
+
+  src = fetchFromGitHub {
+    owner = "ducktors";
+    repo = "turborepo-remote-cache";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-IBydX1+rPgFfgdov7pjUI7gyHBRXColcDaBOmzNNog8=";
+  };
+
+  pnpmDeps = pnpm_10.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    fetcherVersion = 1;
+    hash = "sha256-NlHQ3eJYFvZTGLzfDJPsBVlpMXOTeWdaTUukQQdLA+0=";
+  };
+
+  nativeBuildInputs = [
+    nodejs-slim_22
+    pnpm_10.configHook
+    jq
+    makeWrapper
+  ];
+
+  postPatch = ''
+    # Replace build script to skip linting
+    jq '.scripts.build = "tsc -p ./tsconfig.json"' package.json > package.json.tmp
+    mv package.json.tmp package.json
+
+    # Remove prepare script since we don't need git hooks
+    jq 'del(.scripts.prepare)' package.json > package.json.tmp
+    mv package.json.tmp package.json
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    pnpm build
+    pnpm prune --prod
+    # Clean up broken symlinks left behind by `pnpm prune`
+    # https://github.com/pnpm/pnpm/issues/3645
+    find node_modules -xtype l -delete
```

If you run the the build again with `--check` for the stable `nix-build` CLI or `--rebuild` for the newer `nix build` CLI, you can see that the two outputs differ. To find out why, you can run `diffoscope` with those outputs, and you'll see that PNPM puts there some timestamps and a random storeDir. These files are not needed for the package, we can remove them to make the build reproducible

```suggestion
    find node_modules -xtype l -delete
    # Remove non-deterministic files
    rm node_modules/.modules.yaml
    rm node_modules/.pnpm-workspace-state.json

```

---

When I was making a comment above, I noticed this error. I'll check later why this file doesn't exist when trying to delete it, I'm pretty sure I previously tested it, maybe something changed on master?
```console
       > rm: cannot remove 'node_modules/.pnpm-workspace-state.json': No such file or directory
```